dash-renderer/src/observers/documentTitle.ts

```diff
@@ -0,0 +1,54 @@
+import { IStoreObserverDefinition } from '../StoreObserver';
+import { IStoreState } from '../store';
+
+const updateTitle = (getState: () => IStoreState) => {
+    const {
+        config,
+        isLoading
+    } = getState();
+
+    const update_title = config?.update_title;
+
+    if (!update_title) {
+        return;
+    }
+
+    if (isLoading) {
+        if (document.title !== update_title) {
+            observer.title = document.title;
+            document.title = update_title;
+        }
+    } else {
+        if (document.title === update_title) {
+            document.title = observer.title;
+        } else {
+            observer.title = document.title;
+        }
+    }
+};
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    inputs: ['isLoading'],
+    mutationObserver: undefined,
+    observer: ({
+        getState
+    }) => {
+        const {
+            config
+        } = getState();
+
+        if (observer.config !== config) {
+            observer.config = config;
+            observer.mutationObserver?.disconnect();
+            observer.mutationObserver = new MutationObserver(() => updateTitle(getState));
+            observer.mutationObserver.observe(
+                document.querySelector('title'),
+                { subtree: true, childList: true, attributes: true, characterData: true }
+            );
+        }
+
+        updateTitle(getState);
```

Comment:
Replaces `<DocumentTitle />` and makes it arguably simpler to handle store changes and DOM mutations in a consistent manner as it doesn't need to deal with component lifecycle anymore.

Reply:
This solution should be more robust as it can react to both the DOM changing and the store changing in ways that are not dependant on exact execution timing (the additional wait broke the test/behavior)